include/pybind11/pytypes.h

```diff
@@ -1637,7 +1649,21 @@ class capsule : public object {
         }
     }
 
-    const char *name() const { return PyCapsule_GetName(m_ptr); }
+    const char *name() const {
+        const char *name = PyCapsule_GetName(m_ptr);
+        if (name == nullptr) {
+            if (PyError_Occurred()) {
```

Comment:
```
if (name == nullptr && PyErr_Occurred()) {
```

I'm strongly opposed to copy-pasting this much code. If it wasn't for the error handling, fine, but reimplementing `name` three times ... totally no.

If you change to re-use, this looks good to me.

Best practice: we need tests covering the changed code. I think it's pretty straightforward. Do you know where to start?

Reply:
Sorry will fix this up. And will add tests too. Perhaps this should be converted to a draft until it is ready to minimize distraction.

Reply:
Draft is for before you want feedback.
It's fine with me either way.